### DISCUSSION TITLE
Using pocketbase as a framework didn't serve static contents in pb_public folder

### DISCUSSION THREAD

**Discussion by microflyer** [pocketbase/pocketbase#831] [Q&A] [answered]
Hi Gani,

I am using pocketbase in my hobby project and it is working great. I need to extend some of the functionalities, so I used it as a framework and generated an exe binary based on that. Everything is working except that the binary I built didn't serve the static contents in pb_public folder. Is there any configuration I need do to enable this feature? 

Thank you.

**Comment by ganigeorgiev** [comment, marked as answer]
> Is there any configuration I need do to enable this feature?

Yes. When extending PocketBase its up  to the developer to register the static serving route. You can see how it is done in https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go, but in general you have to register a new route with the `OnBeforeServe` event hook:

```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
    // serves static files from the provided public dir (if exists)
    e.Router.GET("/*", apis.StaticDirectoryHandler(os.DirFS("./pb_public"), false))

    return nil
})
```

**Reply by microflyer** [reply]
Thank you, I tried it and it is working perfectly.
